core/state/access_witness.go

```diff
@@ -144,7 +144,7 @@ func (aw *AccessWitness) TouchTxOriginAndComputeGas(originAddr []byte) {
 
 func (aw *AccessWitness) TouchTxExistingAndComputeGas(targetAddr []byte, sendsValue bool) {
 	aw.touchAddressAndChargeGas(targetAddr, zeroTreeIndex, utils.BasicDataLeafKey, sendsValue, nil)
-	aw.touchAddressAndChargeGas(targetAddr, zeroTreeIndex, utils.CodeHashLeafKey, false, nil)
+	aw.touchAddressAndChargeGas(targetAddr, zeroTreeIndex, utils.CodeHashLeafKey, sendsValue, nil)
```

Comment:
On a simple value-bearing tx to a non-existent account, we weren't including the write-event to the CODE_HASH in the 21000 intrinsic gas. 

This is something that I thought was this way (fixed way) when I [mentioned this](https://github.com/gballet/go-ethereum/pull/508/files#diff-64508d317d86e7a4a5294d76455a5e74148e26883da9e8b4ffc9bbfa3cc8550eR211) yesterday.

Reply:
I think this is incorrect: the name of the function is `TouchTxExistingAndComputeGas` so it refers to an account access. If the account does not exist, that's when it it written, but the rest of the time, it should only be read.

Reply:
This function is only used [here](https://github.com/gballet/go-ethereum/blob/jsign-more-fixes/core/state_transition.go#L399).

Do you mean we should rename it as maybe to a more generic `TouchTxTarget`?

Reply:
Actually, [looking at the spec](https://eips.ethereum.org/EIPS/eip-4762#write-events-1) this write of codeHash shouldn't be given for free. 

So maybe we should revert this PR proposed change; or agree we're proposing a spec change.

cc @tanishqjasoria

Reply:
i am not sure about this
because i remember the discussion about keeping the cost of a normal transfer be 21000, so by that logic we should give the codeHash write for free.

Reply:
> i remember the discussion about keeping the cost of a normal transfer be 21000, so by that logic we should give the codeHash write for free.

But I'm not sure a value-bearing CALL to a non existent account falls into the "normal transfer" definition?

Reply:
> i remember the discussion about keeping the cost of a normal transfer be 21000, so by that logic we should give the codeHash write for free.

So we actually need to add a write-event in that spec link, correct?

Reply:
> (i.e: we have no guarantee the target exists or not)

this is the point I'm making: we should decide if it's to be written once we know if the target exists

> But I'm wondering, apart from 7702, is there any situation where write-touching codeHash for existing accounts makes a gas difference? Since codeHash should be immutable.

that's a good point. Even in 7702, iirc, you don't change the code hash. It's `hash(0xef01)`.

> rename the function?

yeah let's do that, but we need a spec update to go with it

Reply:
> > But I'm wondering, apart from 7702, is there any situation where write-touching codeHash for existing accounts makes a gas difference? Since codeHash should be immutable.
> 
> that's a good point. Even in 7702, iirc, you don't change the code hash. It's `hash(0xef01)`.

From my last question to Matt, it should be `keccak(0xef0100 ++ Addr)` so 7702 actually allows to mutate codeHashes.

I'll rename the function.

Reply:
@gballet  https://github.com/gballet/go-ethereum/pull/511/commits/b13fafc42d98263b89db377d365bc5f214e16558